bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/EnergyApi.java

```diff
@@ -0,0 +1,127 @@
+ *
+ */
+
+@NonNullByDefault
+public class EnergyApi extends RestManager {
+
+    public EnergyApi(ApiBridge apiClient) {
+        super(apiClient, FeatureArea.ENERGY);
+    }
+
+    public NAHome getHomeStatus(String homeId) throws NetatmoException {
+        UriBuilder uriBuilder = getApiUriBuilder(SPATH_HOMESTATUS, PARM_HOMEID, homeId).queryParam(PARM_DEVICETYPE,
+                ModuleType.NAPlug.name(), ModuleType.NRV.name(), ModuleType.NATherm1.name());
+
+        NAHomeStatusResponse response = get(uriBuilder, NAHomeStatusResponse.class);
+        return response.getBody().getHome();
+    }
+
+    /**
+     *
+     * The method switchschedule switches the home&#x27;s schedule to another existing schedule.
+     *
+     * @param homeId The id of home (required)
+     * @param scheduleId The schedule id. It can be found in the getthermstate response, under the keys
+     *            therm_program_backup and therm_program. (required)
+     * @return boolean success
+     * @throws NetatmoException If fail to call the API, e.g. server error or cannot deserialize the
+     *             response body
+     */
+    public boolean switchSchedule(String homeId, String scheduleId) throws NetatmoException {
+        UriBuilder uriBuilder = getAppUriBuilder(SPATH_SWITCHSCHEDULE, PARM_HOMEID, homeId, PARM_SCHEDULEID,
+                scheduleId);
+        post(uriBuilder, ApiResponse.Ok.class, null);
+        return true;
+    }
+
+    /**
+     *
+     * This endpoint permits to control the heating of a specific home. A home can be set in 3 differents modes:
+     * "schedule" mode in which the home will follow the user schedule
+     * "away" mode which will put the whole house to away (default is 12° but can be changed by the user in its
+     * settings)
+     * "hg" corresponds to frostguard mode (7° by default)
+     *
+     * @param homeId The id of home (required)
+     * @param mode The mode. (required)
+     * @return boolean success
+     * @throws NetatmoException If fail to call the API, e.g. server error or cannot deserialize the
+     *             response body
+     */
+    public boolean setThermMode(String homeId, String mode) throws NetatmoException {
+        UriBuilder uriBuilder = getApiUriBuilder(SPATH_SETTHERMMODE, PARM_HOMEID, homeId, PARM_MODE, mode);
+        post(uriBuilder, ApiResponse.Ok.class, null);
+        return true;
+    }
+
+    /**
+     *
+     * The method setroomthermpoint changes the Thermostat manual temperature setpoint.
+     *
+     * @param homeId The id of home (required)
+     * @param roomId The id of the room (required)
+     * @param mode The mode. (required)
+     * @param endtime When using the manual or max setpoint_mode, this parameter defines when the setpoint
+     *            expires. (optional)
+     * @param temp When using the manual setpoint_mode, this parameter defines the temperature setpoint (in
+     *            Celcius) to use. (optional)
```

Comment:
As type is respectively long an ddouble, "optional" has no real sense.

Reply:
Correct